fileio/ft_write_data.m

```diff
@@ -875,12 +875,130 @@ function ft_write_data(filename, dat, varargin)
     data.sampleinfo = [1 size(dat,2)];
     
-    % convert the raw data structure to Homer format
-    nirs = fieldtrip2homer(data, 'event', evt);
+    % devide data in nirs channels, stimulus channels and auxillary channels
+    seldat  = startsWith(hdr.chantype, 'nirs');
+    selstim = strcmp(hdr.chantype, 'stimulus');
+    selaux  = ~seldat & ~selstim;
+    
+    % create empty SnirfClass
+    snirf = SnirfClass();
+    
+    % collect information for creation of snirf file
+    source_idx = find(contains(data.hdr.opto.optotype, {'transmitter', 'source'}));
+    detector_idx = find(contains(data.hdr.opto.optotype, {'receiver', 'detector'}));
+    tra = data.hdr.opto.tra;
+    tra_t = tra'; % transpose tra matrix to get indices of wavelength by row (thus by channel)
+    wl_idx = find(tra_t>0);
+    all_wavelengths = data.hdr.opto.wavelength(tra_t(wl_idx));
+    split = median(all_wavelengths);
+    WL1.values = all_wavelengths(all_wavelengths<split);
+    WL2.values = all_wavelengths(all_wavelengths>split);
+    WL1.nominal = round(median(WL1.values),-1);
+    WL2.nominal = round(median(WL2.values),-1);
+    num_WL = 2;
+    ft_warning('assuming that the nominal wavelengths are %d and %d nm', WL1.nominal, WL2.nominal)
+     
+    % metaDataTags
+    snirf.metaDataTags(1).tags.LengthUnit = data.hdr.opto.unit; 
+    snirf.metaDataTags(1).tags.TimeUnit = 's'; 
+    snirf.metaDataTags(1).tags.FrequencyUnit = 'Hz';
+    
+    % data 
+    % (FIX ME: in theory this can contain multiple data blocks)
+    snirf.data(1).dataTimeSeries = data.trial{1}(seldat,:)'; % <number of time points> x <number of channels>
+    snirf.data(1).time = data.time{1}'; % <number of time points x 1> (can also be  represented as <start time x sample time spacing>
+    
+    % measurementList
+    for i=1:size(tra,1)
+      source = find(tra(i,:)>0);
+      detector = find(tra(i,:)<0);
+      snirf.data.measurementList(i).sourceIndex = find(source_idx==source);
+      snirf.data.measurementList(i).detectorIndex = find(detector_idx==detector);
+%       snirf.data.measurementList(i).wavelengthActual =
+%       all_wavelengths(i); % is not yet supported by the snirf toolbox
+      if any(all_wavelengths(i)==WL1.values)
+        snirf.data.measurementList(i).wavelengthIndex = 1;
+      else
+        snirf.data.measurementList(i).wavelengthIndex = 2;
+      end
+      snirf.data.measurementList(i).dataType = 99999;
+      snirf.data.measurementList(i).dataTypeLabel = 'dOD';
+    end
+    ft_warning('assuming that the input data represents (change in) optical densities')
+    % sort the channels according to wavelengths (because this is the way
+    % that homer handles data) and change the order of the data accordingly
+    [~, idx] = sort(([snirf.data.measurementList(:).wavelengthIndex]));
+    snirf.data.measurementList = snirf.data.measurementList(idx);
+    % update the data accordingly
+    snirf.data.dataTimeSeries=snirf.data.dataTimeSeries(:, idx);
+    
+    % stim
```

Comment:
so I understand that you do not want to store the stim channels that are continuously represented. But what if there is a stim channel, but no evt?

I propose something like

```
if ~isempty(evt) && ~any(selstim)
   % the current code
elseif isempty(evt) && any(selstim)
   % store the stim channels just like the data and aux
elseif ~isempty(evt) && any(selstim)
   % both are present
   % hmm, a decision is needed here -> store both? Or have an key-value option that specifies which one?
else
  % no stim channels and no events
end
```

Reply:
so if there is no evt but stimulus channels, you won't convert them to events, but add them as (aux) channels? Sorry, I don't have much experience with handling stim channels.

Reply:
Sorry, I did not read https://github.com/fNIRS/snirf/blob/master/snirf_specification.md#nirsistimjdata properly. I now see that the `stim` section is a table (a bit like `trl`), not a representation similar to continuously sampled trigger channels. Please ignore this code suggestion.

In general in FT if we use `ft_read_event` to parse continuous trigger channels (i.e. determine the onset of the blocks), then we still represent the trigger channel continuously as well. Throwing away the stim channels means that certain information gets lost, hence I would prefer being able to keep them as aux channels.